### PR TITLE
Abortexit

### PR DESCRIPTION
**Pull request by moneromooo-monero**

### CODE REVIEW

File: src/common/stack_trace.cpp [hunk 1]
```diff
@@ -108,7 +108,9 @@ void set_stack_trace_log(const std::string &log)
   stack_trace_log = log;
 }
 
-void log_stack_trace(const char *msg)
+}  // namespace tools
+
+extern "C" void log_stack_trace(const char *msg)
```
Comment (vtnerd) [inline src/common/stack_trace.cpp]: This probably is not a good right, there are a number of functions that throw in here. This should be undefined behavior if it does happen.

I am not sure why this change was made though ?

Comment (moneromooo-monero) [reply src/common/stack_trace.cpp]: I originally intended to call this before exit from C files, but it turns out the places I wanted to do so are in crypto, which can't use common as a dep since thre's already a dep in the other direction. So I'll drop this instead.

**Comment by moneromooo-monero** [comment]
The exit funcs are in crypto though, can't use tools there.
